### PR TITLE
feat(oxlint): implement `--print-config` cli option

### PR DESCRIPTION
**Pull request by nicksrandall**
I've attempted to implement [`--print-config`](https://eslint.org/docs/latest/use/command-line-interface#--print-config) for oxlint. This implementation works but I'm almost certain there is a better way to do this and I'm open to feedback.

See discussion: #4279

### CODE REVIEW

**Comment by DonIsaac** [comment]
This looks related to #4720. I'd like to reach consensus there before we merge this.

**Comment by Boshen** [comment]
The cli can get merged first because it's a eslint command. 

Behavior wise I think it should just print out whatever is read, it doesn't matter whether it's v8 or v9.

**Comment by DonIsaac** [comment]
If ESLint supports it then we should too.

**Comment by DonIsaac** [comment]
@nicksrandall it appears CI checks are not passing, let me know when you're ready for a review.

**Comment by nicksrandall** [comment]
@DonIsaac I believe I'm ready for review. Thanks you!

**Comment by nicksrandall** [comment]
I'm not sure why the tests are failing... there must be some issue when a struct implements "Serialize" and "JsonSchema" traits? Any idea on how I can fix ?

**Comment by DonIsaac** [comment]
Tests are failing because there are schema snapshots that need updating. Have you run `cargo test` and committed the changes?

File: crates/oxc_linter/src/lib.rs [hunk 1]
```diff
@@ -142,6 +142,24 @@ impl Linter {
...
 
+    /// # Panics
+    pub fn print_config<W: Write>(&self, writer: &mut W) {
+        // merge rules with eslint_config
+        let additional_rules = OxlintRules::from_rules(self.rules.clone());
+        let rules = serde_json::to_value(additional_rules).unwrap();
+        let mut config = serde_json::to_value(self.eslint_config.as_ref()).unwrap();
+        config
+            .get_mut("rules")
+            .unwrap()
```
Comment (DonIsaac) [inline crates/oxc_linter/src/lib.rs:153]: Please use `expect` and provide an error message, otherwise users will have no idea what happened.

File: tasks/website/src/linter/snapshots/schema_markdown.snap [hunk 2]
```diff
@@ -58,153 +56,3 @@ See [Oxlint Rules](./rules)
 
 
 ## settings
-
```
Comment (DonIsaac) [inline tasks/website/src/linter/snapshots/schema_markdown.snap:61]: Why were these removed?

**Comment by mysteryven** [comment]
Hi! @nicksrandall Do you mind me taking this?

**Comment by nicksrandall** [comment]
Please do!

**Comment by mysteryven** [comment]
Close in favor of #6579
